### PR TITLE
[Perf] Skip repeated sampling params validation for streaming input

### PR DESCRIPTION
**Pull request by sherkevin**
## Summary

- avoid re-running full `SamplingParams`/`PoolingParams` validation for streaming input chunks that reuse the request-level sampling params
- keep validation enabled for explicit per-chunk `sampling_params`, so changed chunk-level parameters are still checked before being sent to engine core
- add focused async streaming tests covering both reused request-level params and explicit per-chunk params

## Why this is useful

Streaming input sessions can submit many small chunks while reusing the same request-level `SamplingParams`. The request-level params are already validated when `_add_streaming_input_request()` creates the final sentinel request. Re-validating those same params for every chunk repeats work in the frontend hot path, including logprob/logit-bias/logits-processor/structured-output checks inside `SamplingParams.verify()`.

This change only skips validation for chunks that omit `StreamingInput.sampling_params` and therefore reuse the already-validated request-level params. Explicit per-chunk params continue to validate normally.

## Duplicate-work check

I checked open PRs before opening this:

- `Avoid re-validating reused sampling parameters`: no open PRs
- `streaming SamplingParams validation`: no open PRs
- `validate_params process_inputs`: no open PRs
- `streaming input sampling params`: returned #42433, #37843, #35904; these cover EC transfer params, streaming `max_tokens` session updates, and structured output/reasoning, not repeated params validation
- `streaming input validation`: returned related streaming/realtime PRs such as #34793, but those fix empty-stream/realtime error behavior rather than skipping repeated params validation

No linked issue; this addresses the existing TODO in `vllm/v1/engine/async_llm.py`.

## Tests

- `uvx ruff check vllm/v1/engine/async_llm.py vllm/v1/engine/input_processor.py tests/v1/streaming_input/test_async_llm_streaming.py`
- `uvx ruff format --check vllm/v1/engine/async_llm.py vllm/v1/engine/input_processor.py tests/v1/streaming_input/test_async_llm_streaming.py`
- `python -m pytest tests/v1/streaming_input/test_async_llm_streaming.py -q`

## AI assistance

Used OpenAI Codex to inspect the streaming input path, implement the small patch, run duplicate-work searches, and execute the targeted tests.

### CODE REVIEW

**Comment by sherkevin** [comment]
Closing after re-evaluating the PR against the project contribution policy and the current checks. The remaining failures are gate-related, but the change itself is too small to justify keeping this as a contribution target without stronger evidence/benchmarking.
